app-misc/elasticsearch/elasticsearch-5.1.1.ebuild

```diff
@@ -0,0 +1,84 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit eutils systemd user
+
+MY_PN="${PN%-bin}"
+DESCRIPTION="Open Source, Distributed, RESTful, Search Engine"
+HOMEPAGE="https://www.elastic.co/products/elasticsearch"
+SRC_URI="https://artifacts.elastic.co/downloads/${MY_PN}/${MY_PN}-${PV}.tar.gz"
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RESTRICT="strip"
+
+RDEPEND="|| ( virtual/jre:1.8 virtual/jre:1.7 )"
+
+S="${WORKDIR}/${MY_PN}-${PV}"
+
+pkg_preinst() {
+	if has_version '<app-misc/elasticsearch-2.3.2'; then
+		export UPDATE_NOTES=1
+	fi
+}
+
+pkg_setup() {
+	enewgroup ${MY_PN}
+	enewuser ${MY_PN} -1 /bin/bash /usr/share/${MY_PN} ${MY_PN}
+	esethome ${MY_PN} /usr/share/${MY_PN}
+}
+
+src_prepare() {
+	rm -rf bin/*.{bat,exe}
+	rm LICENSE.txt
+}
+
+src_install() {
+	dodir /etc/${MY_PN}
+	dodir /etc/${MY_PN}/scripts
+
+	insinto /etc/${MY_PN}
+	doins config/*
+	rm -rf config
+
+	insinto /usr/share/${MY_PN}
+	doins -r ./*
+
+	insinto /usr/share/${MY_PN}/bin
+	doins "${FILESDIR}/elasticsearch-systemd-pre-exec"
+
+	chmod +x "${D}"/usr/share/${MY_PN}/bin/*
+
+	keepdir /var/{lib,log}/${MY_PN}
+	keepdir /usr/share/${MY_PN}/plugins
+
+	insinto /usr/lib/tmpfiles.d
+	newins "${FILESDIR}/${MY_PN}.tmpfiles.d" "${MY_PN}.conf"
```

Comment:
Please use the `systemd_newtmpfilesd` eclass function instead: https://devmanual.gentoo.org/eclass-reference/systemd.eclass/index.html